### PR TITLE
Map write support

### LINKED ISSUE #361: [FEATURE] Map data type write
Add write support for map type

### PR DESCRIPTION
**Pull request by Aryex**
### Summary

Implemented Map write support for external tables. Schema checking for map is implemented as well.

### Related Issue

Closes #361 

### Additional Reviewers

@alexey-temnikov 
@alexr-bq 
@jeremyp-bq 
@jonathanl-bq

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala [hunk 1]
```diff
@@ -455,13 +457,31 @@ class SchemaTools extends SchemaToolsInterface {
   override def getVerticaTypeFromSparkType(sparkType: DataType, strlen: Long, arrayLength: Long, metadata: Metadata): SchemaResult[String] = {
     sparkType match {
       // To be reconsidered. Store as binary for now
```
Comment (jeremyp-bq) [inline connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:459]: No longer need this comment.

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala [hunk 2]
```diff
@@ -589,15 +609,23 @@ class SchemaTools extends SchemaToolsInterface {
...
       columnDefs
     }
 
+    def castToVarchar: String => String = colName => colName + "::varchar AS " + addDoubleQuotes(colName)
+
+    def castToArray(colInfo: ColumnDef): String = {
+      val colName = colInfo.label
+      colInfo.childDefinitions.headOption match {
+        case Some(element) => s"($colName::ARRAY[${element.colTypeName}]) as $colName"
+        case None => s"($colName::ARRAY[UKNOWN]) as $colName"
```
Comment (jeremyp-bq) [inline connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:625]: I assume this should be UNKNOWN (missing the first N).

Comment (Aryex) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:625]: Fixed.

**Comment by jeremyp-bq** [comment]
I think we need to update the README to note the Map support.

Should we also add a Map example (and/or Row example) to the examples, or is it not worth it since the support is limited?

**Comment by Aryex** [comment]
> I think we need to update the README to note the Map support.
> 
> Should we also add a Map example (and/or Row example) to the examples, or is it not worth it since the support is limited?

Readme added. As for the example, I also didn't add one for row type. I'll include both in a sperate PR.

**Review by jeremyp-bq — APPROVED** [review]

**Pull request merged**